tests/api/test_mod_ref_type.py

```diff
@@ -105,21 +106,27 @@ def test_destroy_mrt(self, test_mod_ref_type, auth_headers): # noqa
             assert response.status_code == status.HTTP_404_NOT_FOUND
 
     def test_display_order(self, db, test_mod_ref_type, auth_headers): # noqa
-        mod_id = db.query(ModModel.mod_id).filter(ModModel.abbreviation == "ZFIN").one_or_none()
+        # mod_id = db.query(ModModel.mod_id).filter(ModModel.abbreviation == "ZFIN").one_or_none()
+        mod_id = db.query(ModModel.mod_id).filter(ModModel.abbreviation == "ZFIN").scalar()
```

Comment:
In perl scalar(@array) would give you the length of the array.  Is this getting a scalar by extracting the first value, like .one() used to do ?  or is it context dependent ?

Reply:
In SQLAlchemy, the method .scalar() is used to return the first column of the first row of the query result, or None if no rows are found. It is not equivalent to Perl's scalar(@array) which gives the length of an array. The .scalar() in SQLAlchemy is used for when you expect a single value from the query and do not need the full row object.

For example,  mod_id = db.query(ModModel.mod_id).filter(ModModel.abbreviation == "ZFIN").scalar()
we only query **ModModel.mod_id**

   we can rewrite it to :

   mod = db.query(ModModel).filter(ModModel.abbreviation == "ZFIN").first()
   mod_id = mod.mod_id
   
   Does this make sense?

Reply:
Yes, thanks !